scripts/gen_regex.py

```diff
@@ -4,89 +4,74 @@
 from pkg_resources import resource_filename
 
 
+CATEGORIES = [unicodedata.category(chr(i)) for i in range(0x110000)]
 DATA_PATH = pathlib.Path(resource_filename('wordfreq', 'data'))
 
 
+def func_to_regex(func):
+    """
+    Given a function that returns True or False for a numerical codepoint,
+    return a regex character class accepting the characters resulting in True.
+    Ranges separated only by unassigned characters are merged for efficiency.
+    """
+    # A list of [start, end (accepted), end (accepted or unassigned)] lists
+    ranges = []
+
+    for i, cat in enumerate(CATEGORIES):
+        if func(i):
+            # If the last range can be extended, do so; else start a new one
+            if ranges and ranges[-1][2] == i - 1:
+                ranges[-1][1] = i
+                ranges[-1][2] = i
+            else:
+                ranges.append([i, i, i])
+        elif cat == 'Cn':
+            # If the last range can be extended, do so
+            if ranges and ranges[-1][2] == i - 1:
+                ranges[-1][2] = i
+
+    return '[%s]' % ''.join(chr(r[0]) + '-' + chr(r[1]) for r in ranges)
+
+
 def cache_regex_from_func(filename, func):
     """
     Generates a regex from a function that accepts a single unicode character,
     and caches it in the data path at filename.
     """
-    with (DATA_PATH / filename).open(mode='w') as file:
-        file.write(func_to_regex(func))
+    with (DATA_PATH / filename).open(mode='w') as f:
+        f.write(func_to_regex(func))
```

Comment:
...wait, no, my syntax highlighting is wrong (it stlil thinks it's in 2.7).  Let me put this back.